build-logic/src/main/kotlin/modelix-kotlin-jvm.gradle.kts

```diff
@@ -0,0 +1,43 @@
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    kotlin("jvm")
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(11))
+    }
+}
+
+kotlin {
+    jvmToolchain(11)
+    compilerOptions {
+        jvmTarget.set(JvmTarget.JVM_11)
+        freeCompilerArgs.addAll(listOf("-Xjvm-default=all-compatibility"))
+    }
+}
+
+val nonJUnitProjects = setOf(":mps-model-adapters-plugin", ":bulk-model-sync-lib:mps-test")
```

Comment:
Such exclusions are brittle.
How about having a `modelix-kotlin-jvm` not apply `useJUnitPlatform, but having additional convention plugin named `modelix-kotlin-jvm-with-junit-platform`,
that and includes `modelix-kotlin-jvm` applies `useJUnitPlatform`?


tasks.test {
    if (project.path !in nonJUnitProjects) {
        useJUnitPlatform()
    }
}

Reply:
I tried something similar but it seems like convention plugins can't include other convention plugins.

Reply:
Let's at least document that knowledge in the code affected here.

Reply:
Can the affected projects manually revert the configuration applied via the convention plugin?

Reply:
> I tried something similar but it seems like convention plugins can't include other convention plugins.
It seems to work like this: https://github.com/modelix/modelix.core/blob/c82feb9838ee6833e7bae4799f911d7cf83f1cd3/build-logic/src/main/kotlin/modelix-kotlin-jvm-with-junit-platform.gradle.kts#L21

Reply:
Ah interesting. So it works with the string-based declaration. The shorthand notation \`modelix-kotlin-jvm\` does not work.